integrations/firehydrant/client.py

```diff
@@ -0,0 +1,156 @@
+from enum import StrEnum
+from typing import Any, Optional, AsyncGenerator
+import httpx
+from loguru import logger
+from port_ocean.context.event import event
+
+PAGE_SIZE = 50
+
+endpoint_resource_type_mapper = {
+    "environment": "environments",
+    "service": "services",
+    "incident": "incidents",
+    "retrospective": "post_mortems/reports",
+}
+
+
+class CacheKeys(StrEnum):
+    INCIDENT = "incident"
+
+
+class FirehydrantClient:
+    def __init__(self, base_url: str, api_key: str, app_host: str):
+        self.base_url = base_url
+        self.api_key = api_key
+        self.app_host = app_host
+        self.http_client = httpx.AsyncClient(headers=self.api_auth_header)
+
+    @property
+    def api_auth_header(self) -> dict[str, Any]:
+        return {
+            "Authorization": f"{self.api_key}",
+            "Content-Type": "application/json",
+        }
+
+    async def send_api_request(
+        self,
+        endpoint: str,
+        method: str = "GET",
+        query_params: Optional[dict[str, Any]] = None,
+        json_data: Optional[dict[str, Any]] = None,
+    ) -> dict[str, Any]:
+        try:
+            response = await self.http_client.request(
+                method=method,
+                url=f"{self.base_url}/v1/{endpoint}",
+                params=query_params,
+                json=json_data,
+                headers=self.api_auth_header,
+            )
+            response.raise_for_status()
+            return response.json()
+        except httpx.HTTPStatusError as e:
+            logger.error(
+                f"HTTP error with status code: {e.response.status_code} and response text: {e.response.text}"
+            )
+            raise
+
+    async def get_paginated_resource(
+        self, resource_type: str
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info(f"Getting {resource_type} from Firehydrant")
+        endpoint = endpoint_resource_type_mapper.get(resource_type, resource_type)
+
+        pagination_params: dict[str, Any] = {
+            "page": 1,
+            "per_page": PAGE_SIZE,
+        }
+
+        try:
+            while True:
+                response = await self.send_api_request(
+                    endpoint=endpoint, query_params=pagination_params
+                )
+                if response.get("data"):
+                    yield response["data"]
+                else:
+                    logger.warning(f"No {resource_type} found in the response")
```

Comment:
is it possible that the api will return 200 and the response wont have a "data" key in it?
Try to follow the api if something in the api does not work like it should please dont handle it
the reason for this is when we try to write an integration we are trusting the thrid party api to work and not break its interface